app/assets/javascripts/autocomplete.coffee

```diff
@@ -0,0 +1,54 @@
+delay = do ->
+  timer = 0
+  (callback, ms) ->
+    clearTimeout timer
+    timer = setTimeout(callback, ms)
+    return
+
+update_textfield = (list_element) ->
+  $(list_element).removeClass('suggestion-focused')
+  $('.js-autocomplete-textfield').val($(list_element).find('.res-name')[0].innerHTML)
```

Comment:
> What do you think about adding a hidden input with repository_id as value and we can use the value if it's present?

Yeah, I like this idea. By default send `owner/name` but if it's selected via autocomplete also send a hidden `repository_id` and use that if it's supplied. You're going to need to make sure you have tests for this as well.

I'd check with @tarebyte to make sure he likes this plan before implementing.